processing/src/main/java/org/apache/druid/segment/column/ColumnDescriptor.java

```diff
@@ -110,6 +110,24 @@ public ColumnHolder read(
     for (ColumnPartSerde part : parts) {
       part.getDeserializer().read(buffer, builder, columnConfig);
     }
+    return builder.build();
+  }
+
+  public ColumnHolder readProjection(
```

Comment:
I wonder if it would be better to add `@Nullable ColumnHolder parent` to `read`. The amount of copied and tweaked code is high with this separate-method approach.

Reply:
done, consolidated this and deserializer methods